core/services/pipeline/orm.go

```diff
@@ -40,7 +40,7 @@ const KeepersObservationSource = `
                               evmChainID="$(jobSpec.evmChainID)"
                               contract="$(jobSpec.contractAddress)"
                               from="$(jobSpec.fromAddress)"
-                              gas="$(jobSpec.performUpkeepGasLimit)"
+                              gas="$(jobSpec.maxGasLimit)"
```

Comment:
I wonder if it would be safer to modify the ethcall task to support unlimited gas. That way we could avoid edge case failures from the limit being too high. If not, then this might need to be configurable so that we can work around chains with different block size limits.

Reply:
Agree that would be cleaner! We tried sending 0/null here but saw this limit was applied by default: https://docs.chain.link/docs/configuration-variables/#eth_gas_limit_default 

Not sure why that's 500K, maybe we should just have that config as 50M / chain dependent?

Reply:
We have calibrated it for each chain already, but dynamic gas use on chains like Arbitrum create a problem still. When you _send_ a tx through the txm, it takes this in to account for you, so the configured defaults work fine. But when you do an `ethcall`, that doesn't go through the txm, so the same limits don't work the same way. IMHO the simplest solution would be to just avoid the problem by leveraging the 0 == unlimited feature, particularly since that should work everywhere and keep things more flexible and less complicated.

However, if this doesn't work or is not possible, then it would be possible to replicate the txm gas estimation logic here in the pipeline (or wire it in somehow?), but that could get messy and requires a special Arbitrum pipeline.

Reply:
dONE